cmd/network/packet-capture.go

```diff
@@ -0,0 +1,326 @@
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+
+	packetCaptureCmd.Flags().IntVarP(&ops.duration, "duration", "d", packetCaptureDurationSec, "Duration (in seconds) of packet capture")
+	packetCaptureCmd.Flags().StringVarP(&ops.name, "name", "", packetCaptureName, "Name of Daemonset")
+	packetCaptureCmd.Flags().StringVarP(&ops.namespace, "namespace", "n", packetCaptureNamespace, "Namespace to deploy Daemonset")
+	packetCaptureCmd.Flags().StringVarP(&ops.nodeLabelKey, "node-label-key", "", nodeLabelKey, "Node label key")
+	packetCaptureCmd.Flags().StringVarP(&ops.nodeLabelValue, "node-label-value", "", nodeLabelValue, "Node label value")
+
+	ops.startTime = time.Now()
+	return packetCaptureCmd
+}
+
+// packetCaptureOptions defines the struct for running packet-capture command
+type packetCaptureOptions struct {
+	name           string
+	namespace      string
+	nodeLabelKey   string
+	nodeLabelValue string
+	duration       int
+
+	flags *genericclioptions.ConfigFlags
+	genericclioptions.IOStreams
+	kubeCli   client.Client
+	startTime time.Time
+}
+
+func newPacketCaptureOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *packetCaptureOptions {
+	return &packetCaptureOptions{
+		flags:     flags,
+		IOStreams: streams,
+	}
+}
+
+func (o *packetCaptureOptions) complete(cmd *cobra.Command, _ []string) error {
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *packetCaptureOptions) run() error {
+	log.Println("Ensuring Packet Capture Daemonset")
+	ds, err := ensurePacketCaptureDaemonSet(o)
+	if err != nil {
+		log.Fatalf("Error ensuring packet capture daemonset %v", err)
+		return err
+	}
+	log.Println("Waiting For Packet Capture Daemonset")
+	err = waitForPacketCaptureDaemonset(o, ds)
+	if err != nil {
+		log.Fatalf("Error Waiting for daemonset %v", err)
+		return err
+	}
+	log.Println("Copying Files From Packet Capture Pods")
+	err = copyFilesFromPacketCapturePods(o)
+	if err != nil {
+		log.Fatalf("Error copying files %v", err)
+		return err
+	}
+	log.Println("Deleting Packet Capture Daemonset")
+	err = deletePacketCaptureDaemonSet(o, ds)
+	if err != nil {
+		log.Fatalf("Error deleting packet capture daemonset %v", err)
+		return err
+	}
+	return nil
+}
+
+// ensurePacketCaptureDaemonSet ensures the daemonset exists
+func ensurePacketCaptureDaemonSet(o *packetCaptureOptions) (*appsv1.DaemonSet, error) {
+	key := types.NamespacedName{Name: o.name, Namespace: o.namespace}
+	desired := desiredPacketCaptureDaemonSet(o, key)
+	haveDs, current, err := currentPacketCaptureDaemonSet(o, key)
+	if err != nil {
+		log.Fatalf("Error getting current daemonset %v", err)
+		return nil, err
+	}
+
+	if haveDs {
+		log.Println("Already have packet-capture daemonset")
+		return current, nil
```

Comment:
Any reason for reusing the existing DS and not deleting and recreating the DS every time the CLI is executed?

Reply:
Are you suggesting returning an error here instead?

Reply:
Changed to return an error if the daemonset exists.